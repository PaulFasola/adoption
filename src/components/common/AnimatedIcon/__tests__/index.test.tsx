import React from 'react';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import { AnimatedIcon, IconType } from '..';

configure({ adapter: new Adapter() });

describe(`AnimatedIcon`, () => {
  it('Should render', () => {
    Object.keys(IconType).forEach((key) => shallow(<AnimatedIcon type={IconType[key]} />));
  });

  it('Should be styleable', () => {
    const wrapper = shallow(<AnimatedIcon type={IconType.Sucess} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
