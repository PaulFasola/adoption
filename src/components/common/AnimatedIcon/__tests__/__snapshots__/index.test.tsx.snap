// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimatedIcon Should be styleable 1`] = `
<styled.div>
  <styled.svg
    viewBox="0 0 52 52"
    xmlns="http://www.w3.org/2000/svg"
  >
    <styled.circle
      cx="26"
      cy="26"
      fill="none"
      r="25"
    />
    <styled.path
      d="M14.1 27.2l7.1 7.2 16.7-16.8"
      fill="none"
    />
  </styled.svg>
</styled.div>
`;
